apps/ex_nvr_web/test/ex_nvr_web/controllers/api/device_controller_test.exs

```diff
@@ -143,4 +143,24 @@ defmodule ExNVRWeb.API.DeviceControllerTest do
       |> response(404)
     end
   end
+
+  describe "DELETE /api/devices/:id" do
+    setup ctx do
+      %{device: device_fixture(%{settings: %{storage_address: ctx.tmp_dir}})}
+    end
+
+    test "Delete device", %{conn: conn, device: device} do
+      Enum.each(1..10, fn _value -> recording_fixture(device) end)
+      Enum.each(1..5, fn _value -> run_fixture(device) end)
+
+      response =
+        conn
+        |> delete(~p"/api/devices/#{device.id}")
+        |> response(200)
+
+      assert is_nil(Devices.get(device.id))
+      assert length(Recordings.index(device.id)) == 0
+      assert length(Recordings.list_runs(device_id: device.id)) == 0
```

Comment:
You already tested this scenario on `recordings_test.exs`. Just test that the device is deleted and add another test to check the authorization.

And when you want to assert on `nil` or `false` results, you `refute`.